monitor/plugins/spark_progress_upv.py

```diff
@@ -0,0 +1,163 @@
+
+        self.monasca = MonascaMonitor()
+
+        self.submission_url = info_plugin['spark_submisson_url']
+        self.expected_time = info_plugin['expected_time']
+
+        self.remaining_time = int(self.expected_time)
+        self.job_expected_time = int(self.expected_time)
+
+        self.number_of_jobs = int(info_plugin['number_of_jobs'])
+        self.current_job_id = 0
+
+        self.dimensions = {'application_id': self.app_id,
+                           'service': 'spark-sahara'}
+
+        self.conn = paramiko.SSHClient()
+        self.conn.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.conn.connect(hostname=api.mesos_cluster_addr,
+                          username=api.mesos_username,
+                          password=api.mesos_password)
+
+        self.spark_id = self._discover_id_from_spark()
+
+    def _publish_measurement(self, jobs):
+
+        application_progress_error = {}
+
+        # Init
+        jobs.reverse()
+
+        if not len(jobs) == 0:
+            current_job = jobs[self.current_job_id]
+
+            if current_job['status'] == 'FAILED':
+                self.current_job_id = len(jobs) - 1
+
+            elif current_job['status'] == 'SUCCEEDED':
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+
+                self.remaining_time = self.remaining_time - elapsed_time
+
+                self.current_job_id = len(jobs) - 1
+
+                # Job Time
+                self.job_expected_time = (self.remaining_time
+                                 / (float(self.number_of_jobs)
+                                 - float(self.current_job_id)))
+
+            elif current_job['status'] == 'RUNNING':
+                # Job Progress
+                job_progress = (current_job['numCompletedTasks']
+                                / float(current_job['numTasks']))
+
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+
+                # Reference Value
+                ref_value = (elapsed_time / self.job_expected_time)
+
+                # Error
+                error = job_progress - ref_value
+                
+                application_progress_error['name'] = ('application-progress'
+                                                      '.error')
+
+                application_progress_error['value'] = error
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+
+                print application_progress_error['value']
+
+                self.monasca.send_metrics([application_progress_error])
+
+            time.sleep(MONITORING_INTERVAL)
+
+
+    def _get_elapsed_time(self, gmt_timestamp):
+        try:
+            local_tz = tzlocal.get_localzone()
```

Comment:
1. Is this working now? 
2. How are we going to deal with the timezone difference between the spark cluster and the monitor?